src/Features/Core/Portable/IntroduceVariable/IntroduceParameterCodeAction.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Simplification;
+
+namespace Microsoft.CodeAnalysis.IntroduceVariable
+{
+    internal partial class AbstractIntroduceParameterService<TService, TExpressionSyntax, TInvocationExpressionSyntax, TIdentifierNameSyntax>
+    {
+        internal class IntroduceParameterCodeAction : CodeAction
```

Comment:
fwiw, this entire type seems unnecessary.  could be entirely replaced with a normal DocumentChange code action.

Reply:
Removed usages